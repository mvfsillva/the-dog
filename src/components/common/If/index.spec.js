import React from 'react'
import { shallow } from 'enzyme'
import If from '.'

describe('[Component: If]', () => {
  it('Renders children', () => {
    const condition = true
    const component = shallow(
      <If test={condition}>
        <div className="rendered">RENDERED</div>
      </If>,
    )
    expect(component.find('.rendered')).toHaveLength(1)
  })

  it('Does not render children', () => {
    const condition = false
    const component = shallow(
      <If test={condition}>
        <div className="rendered">RENDERED</div>
      </If>,
    )
    expect(component.find('.rendered')).toHaveLength(0)
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<If>If component</If>)
    expect(wrapper).toMatchSnapshot()
  })
})
