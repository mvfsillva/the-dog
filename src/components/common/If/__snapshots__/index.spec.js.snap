// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: If] matches snapshot 1`] = `""`;
